google/cloud/spanner/internal/connection_impl.cc

```diff
@@ -327,6 +321,29 @@ class StreamingPartitionedDmlResult {
   std::unique_ptr<ResultSourceInterface> source_;
 };
 
+std::shared_ptr<spanner::RetryPolicy> const& ConnectionImpl::RetryPolicy()
```

Comment:
There is probably a question about which form we should have used previously, but given that this returns a "prototype" (`shared_ptr`) instead of a "policy" (`unique_ptr`), the commonality of the two interfaces shouldn't be taken too far.  To that end, I've add "prototype" to the function/variable names here, in an attempt to make things clearer.